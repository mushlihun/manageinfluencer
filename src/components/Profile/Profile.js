import React from 'react'
import styled from 'styled-components/macro'
import defaultPicture from '../../icons/default-image.svg'

export default function Profile({
  username,
  categories,
  follower,
  gender,
  birthday,
  location,
  languages,
}) {
  const abbreviateNumber = (n) => {
    if (n < 1e3) return n
    if (n >= 1e3 && n < 1e6) return +(n / 1e3).toFixed(1) + 'K'
    if (n >= 1e6 && n < 1e9) return +(n / 1e6).toFixed(1) + 'M'
    if (n >= 1e9 && n < 1e12) return +(n / 1e9).toFixed(1) + 'B'
    if (n >= 1e12) return +(n / 1e12).toFixed(1) + 'T'
  }
  return (
    <section>
      <ContainerStyled>
        <ImageStyled src={defaultPicture} />
        <UsernameStyled>{username}</UsernameStyled>
        <CategoriesStyled>{categories}</CategoriesStyled>
      </ContainerStyled>
      <DetailListStyled>
        <DetailItemStyled>
          <strong>Follower:</strong> {abbreviateNumber(follower)}
        </DetailItemStyled>
        <DetailItemStyled>
          <strong>Gender:</strong> {gender}
        </DetailItemStyled>
        <DetailItemStyled id="age">
          <strong>Age:</strong> {birthday} years
        </DetailItemStyled>
        <DetailItemStyled>
          <strong>Location:</strong> {location}
        </DetailItemStyled>
        <DetailItemStyled>
          <strong>Languages:</strong> {languages}
        </DetailItemStyled>
      </DetailListStyled>
    </section>
  )
}

const ContainerStyled = styled.section`
  background: var(--transparent-blue);
  height: 250px;
`

const ImageStyled = styled.img`
  width: 140px;
  height: 140px;
  display: block;
  margin-left: auto;
  margin-right: auto;
  padding: 20px 0 0 0;
  border-radius: 50%;
`

const UsernameStyled = styled.h2`
  font-family: 'Poppins';
  font-weight: var(--font-weight-semibold);
  font-size: 25px;
  text-align: center;
  padding: 20px 0 0 0;
  margin-top: 0;
  margin-bottom: 0;
`
const CategoriesStyled = styled.h3`
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  font-size: 20px;
  text-align: center;
  margin-top: 0;
  margin-bottom: 0;
`

const DetailListStyled = styled.ul`
  font-family: 'Poppins';
  font-weight: var(--font-weight-light);
  font-size: 20px;
  text-align: left;
  list-style-type: none;
  padding: 20px;
  width: 250px;
  margin-top: 40px;
  margin-left: auto;
  margin-right: auto;
`
const DetailItemStyled = styled.li`
  letter-spacing: 1px;
  margin: 15px 0;
  position: relative;
`
