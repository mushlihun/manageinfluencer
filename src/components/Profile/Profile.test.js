import React from 'react'
import renderer from 'react-test-renderer'
import { render, getByText } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Profile from './Profile'
import { MemoryRouter } from 'react-router-dom'

describe('Profile', () => {
  it('display your profile', () => {
    const userProfile = [
      {
        id: 1,
        username: '@fitness4life',
        categories: ['Fitness'],
        follower: '10k',
        gender: 'female',
        age: '25 years',
        location: 'Hamburg',
        languages: ['German, ', 'English'],
      },
    ]
    const tree = renderer.create(
      <MemoryRouter>
        <Profile userProfile={userProfile} />
      </MemoryRouter>
    )
    expect(tree).toMatchSnapshot()
  })
})
