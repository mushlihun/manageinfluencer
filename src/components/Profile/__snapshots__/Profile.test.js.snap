// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile display your profile 1`] = `
<section>
  <section
    className="Profile__ContainerStyled-sc-1vgne1u-0 jQaQPJ"
  >
    <img
      className="Profile__ImageStyled-sc-1vgne1u-1 iRYdQZ"
      src="default-image.svg"
    />
    <h2
      className="Profile__UsernameStyled-sc-1vgne1u-2 goUbuk"
    />
    <h3
      className="Profile__CategoriesStyled-sc-1vgne1u-3 fvfCkL"
    />
  </section>
  <ul
    className="Profile__DetailListStyled-sc-1vgne1u-4 kyWEGX"
  >
    <li
      className="Profile__DetailItemStyled-sc-1vgne1u-5 kSFOKB"
    >
      <strong>
        Follower:
      </strong>
       
    </li>
    <li
      className="Profile__DetailItemStyled-sc-1vgne1u-5 kSFOKB"
    >
      <strong>
        Gender:
      </strong>
       
    </li>
    <li
      className="Profile__DetailItemStyled-sc-1vgne1u-5 kSFOKB"
      id="age"
    >
      <strong>
        Age:
      </strong>
       
       years
    </li>
    <li
      className="Profile__DetailItemStyled-sc-1vgne1u-5 kSFOKB"
    >
      <strong>
        Location:
      </strong>
       
    </li>
    <li
      className="Profile__DetailItemStyled-sc-1vgne1u-5 kSFOKB"
    >
      <strong>
        Languages:
      </strong>
       
    </li>
  </ul>
</section>
`;
